llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -4025,11 +4025,8 @@ LoopVectorizationCostModel::computeMaxVF(ElementCount UserVF, unsigned UserIC) {
       MaxPowerOf2RuntimeVF = std::nullopt; // Stick with tail-folding for now.
   }
 
-  if (MaxPowerOf2RuntimeVF && *MaxPowerOf2RuntimeVF > 0) {
-    assert((UserVF.isNonZero() || isPowerOf2_32(*MaxPowerOf2RuntimeVF)) &&
-           "MaxFixedVF must be a power of 2");
-    unsigned MaxVFtimesIC =
-        UserIC ? *MaxPowerOf2RuntimeVF * UserIC : *MaxPowerOf2RuntimeVF;
+  auto ScalarEpilogueNeeded = [this, &UserIC](unsigned MaxVF) {
```

Comment:
The function returns true if no scalar epilogue is needed, so the name should be adjusted?

```suggestion
  auto NoScalarEpilogueNeeded  = [this, &UserIC](unsigned MaxVF) {
```